contexts/MembershipContext/src/Domain/Model/Application.php

```diff
@@ -166,4 +166,16 @@ public function isDeleted(): bool {
 		return $this->isDeleted;
 	}
 
+	public function setFirstPaymentDate( string $firstPaymentDate ) {
+		/** @var PayPalPayment $payPalPayment */
+		$payPalPayment = $this->getPayment()->getPaymentMethod();
+		$payPalData = $payPalPayment->getPayPalData();
+		if ( $payPalData === null ) {
+			$payPalData = new PayPalData();
+		}
+
+		$payPalData->setFirstPaymentDate( $firstPaymentDate );
+		$this->addPayPalData( $payPalData );
```

Comment:
Hi,

this is a very complex dependency tree, there is definitely more to discuss, but given I only took a brief look and the actual scope of the ticket...
+1 for not allowing the empty PayPalData in PayPalPayment (#905). Making it immutable is the logical next step.